sdk/core/core-rest-pipeline/src/retryStrategies/failoverRetryStrategy.ts

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { PipelineRequest } from "../interfaces";
+import { RetryInformation, RetryModifiers, RetryStrategy } from "./retryStrategy";
+import { DEFAULT_CLIENT_MAX_RETRY_INTERVAL, DEFAULT_CLIENT_RETRY_INTERVAL } from "../constants";
+import { DefaultRetryPolicyOptions } from "../policies/defaultRetryPolicy";
+
+/**
+ * A generator for failover hosts. Yields the next host to retry, or yields undefined if failover should be skipped.
+ */
+export type FailoverHostGenerator = (
+  retryState: RetryInformation,
+  options: DefaultRetryPolicyOptions
+) => Generator<FailoverHostState | undefined, void, RetryInformation>;
+
+/**
+ * A host for failover and its associated state.
+ */
+export declare interface FailoverHostState {
```

Comment:
I don't think the `declare` keyword is necessary here or below

Reply:
I learned a new keyword today! Neat! Fixed!